tests/unit_tests/gui/test_main_window.py

```diff
@@ -60,53 +64,134 @@ def opened_main_window(source_root, tmpdir_factory, request):
 
 @pytest.mark.usefixtures("use_tmpdir")
 @pytest.fixture(scope="module")
-def esmda_has_run(opened_main_window, request):
-    qtbot = QtBot(request)
-    gui = opened_main_window
-    qtbot.addWidget(gui)
-
+def esmda_has_run(opened_main_window, run_experiment):
     # Select Multiple Data Assimilation in the simulation panel
-    simulation_panel = gui.findChild(SimulationPanel)
-    assert isinstance(simulation_panel, SimulationPanel)
-    simulation_mode_combo = simulation_panel.findChild(QComboBox)
-    assert isinstance(simulation_mode_combo, QComboBox)
-    current_select = 0
-    simulation_mode_combo.setCurrentIndex(current_select)
-    while simulation_mode_combo.currentText() != MultipleDataAssimilation.name():
-        current_select += 1
+    run_experiment(MultipleDataAssimilation)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@pytest.fixture(scope="module")
+def run_experiment(request, opened_main_window):
+    def func(experiment_mode):
+        qtbot = QtBot(request)
+        gui = opened_main_window
+        qtbot.addWidget(gui)
+        try:
+            shutil.rmtree("poly_out")
+        except FileNotFoundError:
+            pass
+        # Select correct experiment in the simulation panel
+        simulation_panel = gui.findChild(SimulationPanel)
+        assert isinstance(simulation_panel, SimulationPanel)
```

Comment:
Can it end up being something else given that we call `findChild(SimulationPanel` in the previous line?

Reply:
Suppose it can be `None` if we dont find the panel, but this bit of code was just moved around, so cant speak for the original intent. Delete it you think?